src/Maestro/tests/Maestro.ScenarioTests/MaestroScenarioTests.cs

```diff
@@ -0,0 +1,312 @@
+using System;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Maestro.Client;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Octokit;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Maestro.ScenarioTests
+{
+    public class MaestroScenarioTests : IAsyncLifetime
+    {
+        private TestParameters _parameters;
+
+        private readonly ITestOutputHelper _output;
+        private readonly Random _random = new Random();
+
+        public IMaestroApi MaestroApi => _parameters.MaestroApi;
+
+        public GitHubClient GitHubApi => _parameters.GitHubApi;
+
+        public MaestroScenarioTests(ITestOutputHelper output)
+        {
+            _parameters = null!;
+            _output = output;
+        }
+
+        public async Task InitializeAsync()
+        {
+            _parameters = await TestParameters.GetAsync(_output);
+        }
+
+        public Task DisposeAsync()
+        {
+            return Task.CompletedTask;
+        }
+
+        [Fact]
+        public async Task ArcadeSdkUpdate()
+        {
+            var testChannelName = "Test Channel " + _random.Next(int.MaxValue);
+            var sourceRepo = "arcade";
+            var sourceRepoUri = "https://github.com/dotnet/arcade";
+            var sourceBranch = "dependencyflow-tests";
+            var sourceCommit = "0b36b99e29b1751403e23cfad0a7dff585818051";
+            var sourceBuildNumber = _random.Next(int.MaxValue).ToString();
+            var sourceAssets = ImmutableList.Create<AssetData>()
+                .Add(new AssetData(true)
+                {
+                    Name = "Microsoft.DotNet.Arcade.Sdk",
+                    Version = "2.1.0",
+                });
+            var targetRepo = "maestro-test2";
+            var targetBranch = _random.Next(int.MaxValue).ToString();
+            await using var channel = await CreateTestChannelAsync(testChannelName).ConfigureAwait(false);
+            await using var sub = await CreateSubscriptionAsync(testChannelName, sourceRepo, targetRepo, targetBranch, "none");
+            var buildId = await CreateBuildAsync(GetRepoUrl("dotnet", sourceRepo), sourceBranch, sourceCommit, sourceBuildNumber, sourceAssets);
+            await using var _ = await AddBuildToChannelAsync(buildId, testChannelName);
```

Comment:
... what are we disposing here?  We don't care about the variable... this is very confusing code.

Reply:
The AddBuildToChannel thing has cleanup logic to remove the build, thats what the dispose does.